server/lib/puppet-languageserver/puppet_parser_helper.rb

```diff
@@ -1,15 +1,28 @@
 module PuppetLanguageServer
   module PuppetParserHelper
-    def self.remove_char_at(content, line_offsets, line_num, char_num)
+    def self.remove_chars_starting_at(content, line_offsets, line_num, char_num, move_offset)
       line_offset = line_offsets[line_num]
       raise if line_offset.nil?
 
       # Remove the offending character
-      new_content = content.slice(0, line_offset + char_num - 1) + content.slice(line_offset + char_num, content.length - 1)
+      new_content = content.slice(0, line_offset + char_num + move_offset) + content.slice(line_offset + char_num, content.length - 1)
 
       new_content
     end
 
+    def self.remove_char_at(content, line_offsets, line_num, char_num)
+      remove_chars_starting_at(content, line_offsets, line_num, char_num, -1)
+    end
+
+    def self.get_char_at(content, line_offsets, line_num, char_num)
+      line_offset = line_offsets[line_num]
+      raise if line_offset.nil?
+
+      absolute_offset = line_offset + (char_num - 1)
```

Comment:
Internally the `line_num` and `char_num` are zero based (Puppet uses 1 based) so I don't see why you're using `char_num - 1` here.